### PR TITLE
Docs/usage contributing

### LINKED ISSUE #64: [RFE] Create a client getting started guide for developer contributions
**What is the problem you're trying to solve**
Create a general getting started guide for developers

**Describe the solution you'd like**
Create a guide under docs

**Additional context**
Add any other context about the feature request here.

### LINKED ISSUE #98: [RFE] Create basic usage section that could be used for easy copy and pasting
**What is the problem you're trying to solve**
We have Getting Started documentation, but it is not ideal for a quick copy and paste action.

**Describe the solution you'd like**
Create a quick usage section that could be used for copying and pasting commands.

**Additional context**

### PR DESCRIPTION
**Pull request by jpower432**
# Description

- Adds usage docs and usage doc generation (checked in CI during project checks)
- Adds small stylistic/linting code changes for consistency (all covered by unit tests)
- Adds an initial CONTRIBUTING.md

Fixes #64 
Fixes #98 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] This change requires a documentation update

# How Has This Been Tested?

N/A (Docs and styling PR)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Requires updates to the [Getting Started Guide](https://universalreference.io/docs/quick-start/)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Contributing to the UOR Client and Libraries
+
+Thank you for investing your time in contributing to our project!
+
+When contributing to this repository, please first discuss the change you wish to make via issue or GitHub Discussion
```
Comment (sabre1041) [inline CONTRIBUTING.md:5]:
```suggestion
When contributing to this repository, please first discuss the change you wish to make via GitHub Issues or Discussions
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+with the maintainers to ensure the change aligns with the project's long-term plans.
```
Comment (sabre1041) [inline CONTRIBUTING.md:6]:
```suggestion
as to ensure the change aligns with the project's long-term plans.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+### We Use [GitHub Flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+
+Please use the following workflow to make changes to the UOR Client codebase:
+
+1. Fork the repo and create your branch from `main`.
+2. If you've added code that should be tested, add tests.
+3. If you've changed APIs, update the documentation.
+4. Ensure the test suite passes (Run `make test-unit`)
+5. Make sure your code lints (Run `make sanity`)
+6. Create a pull request against the `uor-client-go main` branch.
```
Comment (sabre1041) [inline CONTRIBUTING.md:19]:
```suggestion
6. Create a pull request against the `uor-client-go` `main` branch.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+When applicable, we encourage draft pull requests for early feedback and better communication.
```
Comment (sabre1041) [inline CONTRIBUTING.md:22]:
```suggestion
When applicable, we encourage [draft pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) for early feedback and better communication.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+
+## Report bugs and feature ideas using GitHub's [issues](https://github.com/uor-framework/uor-client-go/issues/new/choose)
+Each issue type has a template attached to guide the submission.
+
+
+## Code Styling
+
+- Run `go fmt`
+- Run `golangci-lint`
```
Comment (sabre1041) [inline CONTRIBUTING.md:36]: Maybe add a link to the project so that the end user can download the required tools

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+3. If you've changed APIs, update the documentation.
+4. Ensure the test suite passes (Run `make test-unit`)
+5. Make sure your code lints (Run `make sanity`)
+6. Create a pull request against the `uor-client-go main` branch.
+
+
+When applicable, we encourage draft pull requests for early feedback and better communication.
+
+*Note: In the Makefile, there are code and file generation targets. If any changes are made to the gRPC API in the `api/services`
+directory, run `make generate-protobuf`. This operation assumes the protobuf compiler and Go plugins are installed. If any changes are made
```
Comment (sabre1041) [inline CONTRIBUTING.md:25]: There are no targets to install some required tools, like `protoc`. Recommend adding some additional documentation on required tooling

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Requested a few changes. Can you also rebase as well as it contains a fix for the timed out builds

File: CONTRIBUTING.md (hunk 5 above)
Comment (jpower432) [reply CONTRIBUTING.md]: Added links to lines.

File: CONTRIBUTING.md (hunk 6 above)
Comment (jpower432) [reply CONTRIBUTING.md:25]: Added a required tools section.

**Review by sabre1041 — APPROVED** [review]

**Pull request merged**